hedera-node/test-clients/src/main/resources/contract/contracts/HederaScheduleService/HederaScheduleService.sol

```diff
@@ -0,0 +1,67 @@
+
+abstract contract HederaScheduleService {
+    address constant scheduleSystemContractAddress = address(0x16b);
+
+    /// Authorizes the calling contract as a signer to the schedule transaction.
+    /// @param schedule the address of the schedule transaction.
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function authorizeSchedule(address schedule) internal returns (int64 responseCode) {
+        (bool success, bytes memory result) = scheduleSystemContractAddress.call(
+            abi.encodeWithSelector(IHederaScheduleService.authorizeSchedule.selector, schedule));
+        responseCode = success ? abi.decode(result, (int64)) : HederaResponseCodes.UNKNOWN;
+    }
+
+    /// Allows for the signing of a schedule transaction given a protobuf encoded signature map
+    /// The message signed by the keys is defined to be the concatenation of the shard, realm, and schedule transaction ID.
+    /// @param schedule the address of the schedule transaction.
+    /// @param signatureMap the protobuf encoded signature map
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function signSchedule(address schedule, bytes memory signatureMap) internal returns (int64 responseCode) {
+        (bool success, bytes memory result) = scheduleSystemContractAddress.call(
+            abi.encodeWithSelector(IHederaScheduleService.signSchedule.selector, schedule, signatureMap));
+        responseCode = success ? abi.decode(result, (int64)) : HederaResponseCodes.UNKNOWN;
+    }
+
+    /// Allows for the creation of a schedule transaction for given a system contract address, abi encoded call data and payer address
+    /// Currently supports the Hedera Token Service System Contract (0x167) with encoded call data for
+    /// createFungibleToken, createNonFungibleToken, createFungibleTokenWithCustomFees, createNonFungibleTokenWithCustomFees
+    /// and updateToken functions
+    /// @param systemContractAddress the address of the system contract from which to create the schedule transaction
+    /// @param callData the abi encoded call data for the system contract function
+    /// @param payer the address of the account that will pay for the schedule transaction
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    /// @return scheduleAddress The address of the newly created schedule transaction.
+    function scheduleNative(address systemContractAddress, bytes memory callData, address payer) internal returns (int64 responseCode, address scheduleAddress) {
+        (bool success, bytes memory result) = scheduleSystemContractAddress.call(
+            abi.encodeWithSelector(IHederaScheduleService.scheduleNative.selector, systemContractAddress, callData, payer));
+        (responseCode, scheduleAddress) = success ? abi.decode(result, (int64, address)) : (int64(HederaResponseCodes.UNKNOWN), address(0));
```

Comment:
Out of curiosity: Why does the response code need a cast to `int64` when it's part of a tuple here and below but not when it isn't above?

Reply:
Because I want to be sure that the type matches the expectation.  It depends on the version of `HederaResponseCodes.sol` one is importing

Reply:
ok ... (that's sad: the bit about depending on which `HederaResponseCodes.sol` you import ...)